File: server/router/mine/mine_projects.go
package mine

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MineProjectsRouter struct{}

func (s *MineProjectsRouter) InitMineProjectsRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mineProjectsRouter := Router.Group("mineProjects").Use(middleware.OperationRecord())
	mineProjectsRouterWithoutRecord := Router.Group("mineProjects")
	mineProjectsRouterWithoutAuth := PublicRouter.Group("mineProjects")
	{
		mineProjectsRouter.POST("createMineProjects", mineProjectsApi.CreateMineProjects)
		mineProjectsRouter.DELETE("deleteMineProjects", mineProjectsApi.DeleteMineProjects)
		mineProjectsRouter.DELETE("deleteMineProjectsByIds", mineProjectsApi.DeleteMineProjectsByIds)
		mineProjectsRouter.PUT("updateMineProjects", mineProjectsApi.UpdateMineProjects)
	}
	{
		mineProjectsRouterWithoutRecord.GET("findMineProjects", mineProjectsApi.FindMineProjects)
		mineProjectsRouterWithoutRecord.GET("getMineProjectsList", mineProjectsApi.GetMineProjectsList)
	}
	{
		_ = mineProjectsRouterWithoutAuth
	}
}

File: web/src/api/mine/mine_projects.js
import service from '@/utils/request'

export const createMineProjects = (data) => {
  return service({
    url: '/mineProjects/createMineProjects',
    method: 'post',
    data
  })
}

export const deleteMineProjects = (params) => {
  return service({
    url: '/mineProjects/deleteMineProjects',
    method: 'delete',
    params
  })
}

export const deleteMineProjectsByIds = (params) => {
  return service({
    url: '/mineProjects/deleteMineProjectsByIds',
    method: 'delete',
    params
  })
}

export const updateMineProjects = (data) => {
  return service({
    url: '/mineProjects/updateMineProjects',
    method: 'put',
    data
  })
}

export const findMineProjects = (params) => {
  return service({
    url: '/mineProjects/findMineProjects',
    method: 'get',
    params
  })
}

export const getMineProjectsList = (params) => {
  return service({
    url: '/mineProjects/getMineProjectsList',
    method: 'get',
    params
  })
}
